### DISCUSSION TITLE
No JWT in found on the request in getServerSideProps of Next.js

### DISCUSSION THREAD

**Discussion by ExpressGradient** [supabase/supabase#3306] [Questions] [answered]
Hey supabase community, I'm using Next.js with Supabase. So, the docs mentioned using `getUserByCookie()` to get User server side. But I can't seem to find any cookie on the request.

Also no JWT on the headers too.

Thanks.

**Comment by silentworks** [comment]
Can't really tell what you are doing without seeing your project, but have a look at https://github.com/dabit3/supabase-nextjs-auth/blob/main/pages/protected.js and you will see how the `getUserByCookie` works, you can also clone that project and see it working there. Maybe you are missing a step in your project and hopefully, this will help. The JWT will be in the cookie, not the headers.

**Reply by ExpressGradient** [reply]
Thanks for replying.

Yes, I thought that the cookie would be set on the client-side when the auth happens automatically. So whenever you do getUserByCookie(). You'd get the user.

But I didn't know that I have to manually set the cookie.

**Reply by ExpressGradient** [reply]
I looked at this nextjs example, https://github.com/vercel/next.js/tree/canary/examples/with-supabase-auth-realtime-db

**Comment by ExpressGradient** [comment, marked as answer]
If anyone faces this issue, it's because you have to set the cookie using `setAuthCookie()` in an API route and call the route in the `_app.js` file. For more info look at this example, https://github.com/vercel/next.js/tree/canary/examples/with-supabase-auth-realtime-db
